### DISCUSSION TITLE
Anvil Contract Address

### DISCUSSION THREAD

**Discussion by luiscaceresd** [Cyfrin/foundry-full-course-cu#1313] [General]
Hi everyone I got a question. I'm doing the course at: https://youtu.be/sas02qSFZ74?t=10177 and I don't understand why the contract I deploy on my Anvil network has the same address as the one Patrick deploys, why is this contract a constant?

```javascript
export const contractAddress = "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512"
```

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/74666044/4ef0ec08-b77b-42a1-bda6-14b8c0c2870e)

Thanks in advance

**Comment by ghost** [comment]
Contract addresses are **deterministically** derived from the deployer address and the number of transactions the deployer has sent (a.k.a nonce). Both values are RLP encoded and then hashed using `keccak256`.

Since the accounts from anvil are shared by all of us, they will produce the same contract addresses over and over again.
